src/test/java/use_case/DeletePostInteractorTest.java

```diff
@@ -0,0 +1,154 @@
+    void deletePostFailsWhenPostDoesNotExistTest() {
+        String postId = "post123";
+        String userId = "user123";
+
+        User mockUser = createMockUser(userId);
+        when(mockUserRepo.getCurrentUser()).thenReturn(mockUser);
+        when(mockPostDB.existsByID(postId)).thenReturn(false);
+
+        DeletePostInputData inputData = new DeletePostInputData(postId, userId);
+
+        DeletePostFailedException exception = assertThrows(DeletePostFailedException.class,
+                () -> interactor.deletePost(inputData));
+        assertEquals("Post with given ID doesn't exist.", exception.getMessage());
+
+        verify(mockPresenter).prepareFailView("Post with given ID doesn't exist.");
+        verify(mockPostDB, never()).deletePost(anyString());
+    }
+
+    @Test
+    void deletePostFailsWhenUserUnauthorizedTest() {
+        String postId = "post123";
+        String userId = "user123";
+
+        User mockUser = createMockUser("otherUser"); // Different user
+        when(mockUserRepo.getCurrentUser()).thenReturn(mockUser);
+        when(mockPostDB.existsByID(postId)).thenReturn(true);
+
+        DeletePostInputData inputData = new DeletePostInputData(postId, userId);
+
+        DeletePostFailedException exception = assertThrows(DeletePostFailedException.class, () -> interactor.deletePost(inputData));
+        assertEquals("User does not have permission to delete this post.", exception.getMessage());
+
+        verify(mockPresenter).prepareFailView("User does not have permission to delete this post.");
+        verify(mockPostDB, never()).deletePost(anyString());
+    }
+
+    @Test
+    void deletePostFailsOnExceptionTest() {
+        String postId = "post123";
+        String userId = "user123";
+
+        User mockUser = createMockUser(userId, List.of(postId));
+        when(mockUserRepo.getCurrentUser()).thenReturn(mockUser);
+        when(mockPostDB.existsByID(postId)).thenReturn(true);
+        doThrow(new RuntimeException("DB error")).when(mockPostDB).deletePost(postId);
+
+        DeletePostInputData inputData = new DeletePostInputData(postId, userId);
+
+        DeletePostFailedException exception = assertThrows(DeletePostFailedException.class, () -> interactor.deletePost(inputData));
+        assertEquals("Failed to delete the post.", exception.getMessage());
+
+        verify(mockPresenter).prepareFailView("Failed to delete the post.");
+        verify(mockPostDB).deletePost(postId);
+    }
+
+    @Test
+    void canDeleteTest() {
+        String postId = "post123";
+        String userId = "user123";
+
+        User mockUser = createMockUser(userId, List.of(postId));
+        when(mockUserRepo.getCurrentUser()).thenReturn(mockUser);
+
+        DeletePostInputData inputData = new DeletePostInputData(postId, userId);
+
+        boolean canDelete = interactor.canDelete(inputData);
+
+        assertTrue(canDelete);
+        verify(mockUserRepo).getCurrentUser();
+    }
+
+    @Test
+    void cannotDeleteTest() {
```

Comment:
What is the different between this test and the deletePostFailsWhenUserUnauthorizedTest ?